CRM/Odoosync/Hook/Post/Contact/Checker.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * Check if contact 'sync info' is already update
+ */
+class CRM_Odoosync_Hook_Post_Contact_Checker {
```

Comment:
I don't see a reason to implement this class, you can just move the implementation  to CRM_Odoosync_Hook_Post_Contact_Base class